app/models/legacy_identifier.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class LegacyIdentifier < ApplicationRecord
+  belongs_to :resource, polymorphic: true
+
+  def self.find_uuid(version:, old_id:)
+    legacy_id = find_by(version: version, old_id: old_id)
+
+    legacy_id&.resource&.uuid ||
+      raise(ActiveRecord::RecordNotFound)
+  end
```

Comment:
Code Climate seems to think this is untested, even though it totally is `spec/models/legacy_identifier_spec.rb`